python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -61,193 +82,140 @@ def make_dagster_job_from_airflow_dag(
+            else:
+                raise DagsterInvariantViolationError(
+                    'Could not find "{AIRFLOW_EXECUTION_DATE_STR}" in tags "{tags}". Please '
+                    'add "{AIRFLOW_EXECUTION_DATE_STR}" to tags before executing'.format(
+                        AIRFLOW_EXECUTION_DATE_STR=AIRFLOW_EXECUTION_DATE_STR,
+                        tags=run_tags,
+                    )
                 )
 
-        Use RepositoryDefinition as usual, for example:
-        ``dagit -f path/to/make_dagster_repo.py -n make_repo_from_dir``
-
-    Args:
-        dag_path (str): Path to directory or file that contains Airflow Dags
-        include_examples (bool): True to include Airflow's example DAGs. (default: False)
-        safe_mode (bool): True to use Airflow's default heuristic to find files that contain DAGs
-            (ie find files that contain both b'DAG' and b'airflow') (default: True)
-        use_airflow_template_context (bool): If True, will call get_template_context() on the
-            Airflow TaskInstance model which requires and modifies the DagRun table. The use_airflow_template_context
-            setting is ignored if use_ephemeral_airflow_db is True.
-            (default: False)
-        mock_xcom (bool): If True, dagster will mock out all calls made to xcom, features that
-            depend on xcom may not work as expected. (default: False)
-        use_ephemeral_airflow_db (bool): If True, dagster will create an ephemeral sqlite airflow
-            database for each run. (default: False)
-        connections (List[Connection]): List of Airflow Connections to be created in the Ephemeral
-            Airflow DB, if use_emphemeral_airflow_db is False this will be ignored.
-
-    Returns:
-        Definitions
-    """
-    check.str_param(dag_path, "dag_path")
-    check.bool_param(safe_mode, "safe_mode")
-    check.bool_param(use_airflow_template_context, "use_airflow_template_context")
-    mock_xcom = check.opt_bool_param(mock_xcom, "mock_xcom")
-    use_ephemeral_airflow_db = check.opt_bool_param(
-        use_ephemeral_airflow_db, "use_ephemeral_airflow_db"
+            dagrun = dag.get_dagrun(execution_date=execution_date)
+            if not dagrun:
+                if is_airflow_2():
+                    dagrun = dag.create_dagrun(
+                        state=DagRunState.RUNNING,
+                        execution_date=execution_date,
+                        run_type=DagRunType.MANUAL,
+                    )
+                else:
+                    dagrun = dag.create_dagrun(
+                        run_id=f"dagster_airflow_run_{execution_date}",
+                        state=State.RUNNING,
+                        execution_date=execution_date,
+                    )
+        return {
+            "dag": dag,
+            "dagrun": dagrun,
+        }
+
+    dag_resource = ResourceDefinition.hardcoded_resource(value=dag, description="DAG model")
```

Comment:
this is super, super confusing. Dag is passed in, the reassigned to be from context.resources.dag, then passed in as a hard-coded resource?

Reply:
I broke this into three resources `airflow_db`, `dag`, `dagrun` i think it makes it a little clearer but tbh the only one that will likely ever be configurable is `airflow_db`.